graphite_api/readers.py

```diff
@@ -13,9 +13,16 @@ def get_intervals(self):
             interval_sets.extend(node.intervals.intervals)
         return IntervalSet(sorted(interval_sets))
 
-    def fetch(self, startTime, endTime):
+    def fetch(self, startTime, endTime, now=None, requestContext=None):
         # Start the fetch on each node
-        results = [n.fetch(startTime, endTime) for n in self.nodes]
+        results = []
+
+        for node in self.nodes:
+            try:
+                results.append(node.fetch(startTime, endTime, now,
+                                          requestContext))
+            except:
```

Comment:
`except Exception:` & `logger.error("fetch error", exc_info=True)` to avoid silently hiding failures?